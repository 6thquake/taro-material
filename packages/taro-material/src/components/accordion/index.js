import Taro from '@tarojs/taro'
import PropTypes from 'prop-types'
import classNames from 'classnames'
import { View, Text } from '@tarojs/components'
import AtComponent from '../../common/component'
import { delayQuerySelector, initTestEnv } from '../../common/utils'

initTestEnv()

// 文档
export default class AtAccordion extends AtComponent {
  constructor () {
    super(...arguments)
    this.isCompleted = true
    this.startOpen = false
    this.state = {
      wrapperHeight: ''
    }
  }

  handleClick = event => {
    const { open } = this.props
    if (!this.isCompleted) return

    this.props.onClick(!open, event)
  }

  toggleWithAnimation () {
    const { open, isAnimation } = this.props
    if (!this.isCompleted || !isAnimation) return

    this.isCompleted = false
    delayQuerySelector(this, '.at-accordion__body', 0)
      .then(rect => {
        const height = parseInt(rect[0].height)
        const startHeight = open ? height : 0
        const endHeight = open ? 0 : height
        this.startOpen = false
        this.setState({
          wrapperHeight: startHeight
        }, () => {
          setTimeout(() => {
            this.setState({
              wrapperHeight: endHeight
            }, () => {
              setTimeout(() => {
                this.isCompleted = true
                this.setState({})
              }, 700)
            })
          }, 100)
        })
      })
  }

  componentWillReceiveProps (nextProps) {
    if (nextProps.open !== this.props.open) {
      this.startOpen = nextProps.open && nextProps.isAnimation
      this.toggleWithAnimation()
    }
  }

  render () {
    const {
      customStyle,
      className,
      title,
      icon,
      hasBorder,
      open
    } = this.props
    const { wrapperHeight } = this.state

    const rootCls = classNames('at-accordion', className)
    const iconCls = classNames({
      'at-icon': true,
      [`at-icon-${icon && icon.value}`]: icon && icon.value,
      'at-accordion__icon': true,
    })
    const headerCls = classNames('at-accordion__header', {
      'at-accordion__header--noborder': !hasBorder
    })
    const arrowCls = classNames('at-accordion__arrow', {
      'at-accordion__arrow--folded': !!open
    })
    const contentCls = classNames('at-accordion__content', {
      'at-accordion__content--inactive': (!open && this.isCompleted) || this.startOpen
    })
    const iconStyle = {
      color: (icon && icon.color) || '',
      fontSize: (icon && `${icon.size}px`) || '',
    }
    const contentStyle = { height: `${wrapperHeight}px` }

    if (this.isCompleted) {
      contentStyle.height = ''
    }

    return <View className={rootCls} style={customStyle}>
      <View className={headerCls} onClick={this.handleClick}>
        {icon && icon.value && <Text className={iconCls} style={iconStyle}></Text>}
        <View className='at-accordion__title'>{title}</View>
        <View className={arrowCls}>
          <Text className='at-icon at-icon-chevron-down'></Text>
        </View>
      </View>
      <View style={contentStyle} className={contentCls}>
        <View className='at-accordion__body'>
          {this.props.children}
        </View>

      </View>
    </View>
  }
}

AtAccordion.defaultProps = {
  open: false,
  customStyle: '',
  className: '',
  title: '',
  icon: {},
  hasBorder: true,
  isAnimation: true,
  onClick: () => {},
}

AtAccordion.propTypes = {
  customStyle: PropTypes.oneOfType([
    PropTypes.object,
    PropTypes.string
  ]),
  className: PropTypes.oneOfType([
    PropTypes.array,
    PropTypes.string
  ]),
  open: PropTypes.bool,
  isAnimation: PropTypes.bool,
  title: PropTypes.string,
  icon: PropTypes.object,
  hasBorder: PropTypes.bool,
  onClick: PropTypes.func,
}
