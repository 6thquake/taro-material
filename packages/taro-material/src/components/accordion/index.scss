@import '../../style/theme/default.scss';
@import '../../style/mixins/index.scss';

.at-accordion {
  &__header {
    position: relative;
    display: flex;
    box-sizing: border-box;
    width: 100%;
    min-height: $font-size-lg;
    font-size: $font-size-lg;
    padding: $spacing-v-lg $spacing-h-lg;
    @include hairline-bottom();
    @include active;
  }

  &__title {
    flex: 1;
    text-align: left;
    @include line;
  }

  &__content {
    position: relative;
    width: 100%;
    transition: height 0.3s;
    overflow: hidden;
    @include hairline-top();
  }
}
