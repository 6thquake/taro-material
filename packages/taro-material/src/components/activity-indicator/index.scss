@import "../../style/mixins/index.scss";
@import "../../style/theme/default.scss";

.at-activity-indicator {
  @include display-flex();

  line-height: $line-height-base;

  &--center {
    @include absolute-center;
  }

  &__body {
    @include flex(0, 0, auto);

    line-height: 0;
  }

  &__content {
    @include flex(0, 0, auto);
    @include align-self(center);

    font-size: 28px;
    margin-left: $spacing-h-lg;
    color: $color-grey-2;
  }
}
