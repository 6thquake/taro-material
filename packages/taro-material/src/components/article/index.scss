@import '../../style/theme/default.scss';

.at-article {
  word-break: break-all;
  word-wrap: break-word;
  line-height: $line-height-zh;

  &__h1,
  &__h2,
  &__h3 {
    color: $color-grey-0;
    margin: 0 30px;
  }

  &__h1 {
    margin-top: 30px;
    font-size: 48px;
    line-height: 72px;
  }

  &__info {
    font-size: $font-size-base;
    color: $color-grey-3;
    line-height: 42px;
    margin: 0 30px;
  }

  &__h2 {
    font-size: $font-size-xxl;
    line-height: 60px;
    margin-top: 30px;
  }

  &__h3 {
    font-size: $font-size-lg;
    line-height: 48px;
  }

  &__p {
    font-size: $font-size-base;
    color: $color-grey-1;
    line-height: 42px;
    margin: 25px 30px 0;
  }

  &__img {
    width: 690px;
    display: block;
    margin: 20px auto 0;
    border-radius: 4px;
  }
}
