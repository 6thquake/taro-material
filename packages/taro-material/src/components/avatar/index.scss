@import '../../style/theme/default.scss';

$avatar-size-sm: 80px;
$avatar-size-md: 100px;
$avatar-size-lg: 120px;

.at-avatar {
  width: $avatar-size-md;
  height: $avatar-size-md;
  color: #fff;
  font-size: $font-size-xxl;
  line-height: $avatar-size-md;
  text-align: center;
  border-radius: $border-radius-md;
  background: $color-grey-4;
  box-shadow: 0 0 50px 0 rgba(0, 0, 0, 0.05);
  overflow: hidden;

  &__img {
    width: 100%;
    height: 100%;
  }

  &--large {
    width: $avatar-size-lg;
    height: $avatar-size-lg;
    font-size: 48px;
    line-height: $avatar-size-lg;
  }

  &--small {
    width: $avatar-size-sm;
    height: $avatar-size-sm;
    font-size: $font-size-lg;
    line-height: $avatar-size-sm;
  }

  &--circle {
    border-radius: 50%;
    background-clip: border-box;
    overflow: hidden;
  }
}
