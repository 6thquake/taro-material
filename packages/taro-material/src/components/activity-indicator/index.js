import Taro from '@tarojs/taro'
import { View, Text } from '@tarojs/components'

import PropTypes from 'prop-types'
import classNames from 'classnames'

import AtLoading from '../loading/index'
import AtComponent from '../../common/component'

import './index.scss'

export default class AtActivityIndicator extends AtComponent {
  render () {
    const { color, size, mode, content } = this.props

    const rootClass = classNames(
      'at-activity-indicator',
      {
        'at-activity-indicator--center': mode === 'center'
      },
      this.props.className
    )

    return (
      <View className={rootClass}>
        <View className='at-activity-indicator__body'>
          <AtLoading size={size} color={color} />
        </View>
        {content && (
          <Text className='at-activity-indicator__content'>{content}</Text>
        )}
      </View>
    )
  }
}

AtActivityIndicator.defaultProps = {
  size: 24,
  color: '#6190E8'
}

AtActivityIndicator.propTypes = {
  size: PropTypes.number,
  mode: PropTypes.string,
  color: PropTypes.string,
  content: PropTypes.string
}
