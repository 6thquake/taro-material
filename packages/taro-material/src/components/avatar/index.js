import Taro from '@tarojs/taro'
import { View, Image, Text, OpenData } from '@tarojs/components'
import PropTypes from 'prop-types'
import classNames from 'classnames'

import AtComponent from '../../common/component'

import './index.scss'

const SIZE_CLASS = {
  large: 'large',
  normal: 'normal',
  small: 'small',
}

export default class AtAvatar extends AtComponent {
  constructor () {
    super(...arguments)
    this.state = {
      isWEAPP: Taro.getEnv() === Taro.ENV_TYPE.WEAPP,
    }
  }

  onClick () {
  }

  render () {
    const {
      size,
      circle,
      image,
      text,
      openData,
      customStyle,
    } = this.props
    const rootClassName = ['at-avatar']

    const classObject = {
      [`at-avatar--${SIZE_CLASS[size]}`]: SIZE_CLASS[size],
      'at-avatar--circle': circle,
    }

    let letter = ''
    if (text) letter = text[0]

    let elem
    if (openData && openData.type === 'userAvatarUrl' && this.state.isWEAPP) {
      elem = (<OpenData type={openData.type}></OpenData>)
    } else if (image) {
      elem = (<Image className='at-avatar__img' src={image} />)
    } else {
      elem = (<Text className='at-avatar__text'>{letter}</Text>)
    }
    return (
      <View
        className={classNames(rootClassName, classObject, this.props.className)}
        style={customStyle}
      >{elem}</View>
    )
  }
}

AtAvatar.defaultProps = {
  size: 'normal',
  circle: false,
  text: '',
  image: '',
  openData: {},
  customStyle: {},
  className: '',
}

AtAvatar.propTypes = {
  size: PropTypes.oneOf(['large', 'normal', 'small']),
  circle: PropTypes.bool,
  text: PropTypes.string,
  image: PropTypes.string,
  openData: PropTypes.object,
  customStyle: PropTypes.oneOfType([PropTypes.object, PropTypes.string]),
  className: PropTypes.oneOfType([PropTypes.array, PropTypes.string]),
}
