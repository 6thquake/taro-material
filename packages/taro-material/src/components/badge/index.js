import PropTypes from 'prop-types'
import Taro from '@tarojs/taro'
import { View } from '@tarojs/components'
import isNaN from 'lodash/isNaN'
import classNames from 'classnames'

import AtComponent from '../../common/component'
import './index.scss'


export default class AtBadge extends AtComponent {
  constructor () {
    super(...arguments)
    this.state = {}
  }

  formatValue (value, maxValue) {
    if (value === '' || value === null) return ''
    const numValue = +value
    if (isNaN(numValue)) {
      return value
    }
    return numValue > maxValue ? `${maxValue}+` : numValue
  }

  render () {
    const {
      dot,
      value,
      maxValue,
      customStyle,
    } = this.props
    const rootClassName = ['at-badge']

    const val = this.formatValue(value, maxValue)

    return (
      <View
        className={classNames(rootClassName, this.props.className)}
        style={customStyle}
      >
        {this.props.children}
        {dot ? <View className='at-badge__dot'></View> : val !== '' && <View className='at-badge__num'>{val}</View>}
      </View>
    )
  }
}

AtBadge.defaultProps = {
  dot: false,
  value: '',
  maxValue: 99,
  customStyle: {},
}

AtBadge.propTypes = {
  dot: PropTypes.bool,
  value: PropTypes.oneOfType([
    PropTypes.string,
    PropTypes.number
  ]),
  maxValue: PropTypes.number,
  customStyle: PropTypes.oneOfType([PropTypes.object, PropTypes.string]),
}
