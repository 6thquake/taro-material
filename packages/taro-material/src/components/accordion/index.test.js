import Nerv from 'nervjs'
import { View } from '@tarojs/components'
import { renderToString } from 'nerv-server'

import AtAccordion from '../../../.temp/components/accordion'

describe('AtAccordion Snap', () => {
  it('render initial AtAccordion', () => {
    const component = renderToString(
      <AtAccordion isTest>
      </AtAccordion>
    )
    expect(component).toMatchSnapshot()
  })

  it('render AtAccordion -- props title', () => {
    const component = renderToString(
      <AtAccordion isTest title='title'>
      </AtAccordion>
    )
    expect(component).toMatchSnapshot()
  })

  it('render AtAccordion -- props open', () => {
    const component = renderToString(
      <AtAccordion isTest open>
      </AtAccordion>
    )
    expect(component).toMatchSnapshot()
  })

  it('render AtAccordion -- props icon', () => {
    const component = renderToString(
      <AtAccordion isTest icon={{ value: 'chevron-down', color: 'red' }}>
        <View></View>
      </AtAccordion>
    )
    expect(component).toMatchSnapshot()
  })
})
