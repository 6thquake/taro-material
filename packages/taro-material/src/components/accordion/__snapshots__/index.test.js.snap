// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AtAccordion Snap render AtAccordion -- props icon 1`] = `"<div class=\\"at-accordion\\"><div class=\\"at-accordion__header\\"><span style=\\"font-size:24px;color:red;margin-right:15px;\\" class=\\"taro-text at-icon at-icon-chevron-down at-accordion__icon\\"></span><div class=\\"at-accordion__title\\"></div><span style=\\"font-size:24px;color:#c7c7cc;transition:transform 0.2s ease;transform:rotate(0);\\" class=\\"taro-text at-icon at-icon-chevron-down\\"></span></div><div style=\\"height:px;\\" class=\\"at-accordion__content\\"><div class=\\"\\"></div></div></div>"`;

exports[`AtAccordion Snap render AtAccordion -- props open 1`] = `"<div class=\\"at-accordion\\"><div class=\\"at-accordion__header\\">undefined<div class=\\"at-accordion__title\\"></div><span style=\\"font-size:24px;color:#c7c7cc;transition:transform 0.2s ease;transform:rotate(180deg);\\" class=\\"taro-text at-icon at-icon-chevron-down\\"></span></div><div style=\\"height:px;\\" class=\\"at-accordion__content\\"></div></div>"`;

exports[`AtAccordion Snap render AtAccordion -- props title 1`] = `"<div class=\\"at-accordion\\"><div class=\\"at-accordion__header\\">undefined<div class=\\"at-accordion__title\\">title</div><span style=\\"font-size:24px;color:#c7c7cc;transition:transform 0.2s ease;transform:rotate(0);\\" class=\\"taro-text at-icon at-icon-chevron-down\\"></span></div><div style=\\"height:px;\\" class=\\"at-accordion__content\\"></div></div>"`;

exports[`AtAccordion Snap render initial AtAccordion 1`] = `"<div class=\\"at-accordion\\"><div class=\\"at-accordion__header\\">undefined<div class=\\"at-accordion__title\\"></div><span style=\\"font-size:24px;color:#c7c7cc;transition:transform 0.2s ease;transform:rotate(0);\\" class=\\"taro-text at-icon at-icon-chevron-down\\"></span></div><div style=\\"height:px;\\" class=\\"at-accordion__content\\"></div></div>"`;
